user/models.py

```diff
@@ -1,5 +1,57 @@
-from django.contrib.auth.models import AbstractUser
+from __future__ import annotations
+
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+from django.db import models
+from django.utils.translation import gettext as _
+
+
+class UserManager(BaseUserManager):
+    """Define a model manager for User model with no username field."""
+
+    use_in_migrations = True
+
+    def _create_user(self, email: str, password: str, **extra_fields) -> User:
+        """Create and save a User with the given email and password."""
+
+        if not email:
+            raise ValueError("The given email must be set")
+
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(
+            self, email: str, password: str = None, **extra_fields
```

Comment:
add typehint for extra_fields
use typehint like: Optional[YOUR_VALUE_TYPE] = None